src/ParallelAlgorithms/Events/MonitorMemory.hpp

```diff
@@ -213,128 +161,177 @@ MonitorMemory<Dim, ObservationValueTag>::MonitorMemory(CkMigrateMessage* msg)
     : Event(msg) {}
 
 template <size_t Dim, typename ObservationValueTag>
-template <typename DbTags, typename Metavariables, typename ArrayIndex,
+template <typename Metavariables>
+MonitorMemory<Dim, ObservationValueTag>::MonitorMemory(
+    const std::optional<std::vector<std::string>>& components_to_monitor,
+    const Options::Context& context, Metavariables /*meta*/) {
+  using component_list = tmpl::push_back<typename Metavariables::component_list,
+                                         Parallel::GlobalCache<Metavariables>>;
+  std::unordered_map<std::string, std::string> existing_components{};
+  std::string str_component_list{};
+
+  tmpl::for_each<component_list>(
+      [&existing_components, &str_component_list](auto component_v) {
+        using component = tmpl::type_from<decltype(component_v)>;
+        const std::string component_name = pretty_type::name<component>();
+        const std::string chare_type_name =
+            pretty_type::name<typename component::chare_type>();
+        existing_components[component_name] = chare_type_name;
+        // Only Array we can monitor is DgElementArray
+        if (chare_type_name != "Array") {
+          str_component_list += " - " + component_name + "\n";
+        } else if (component_name == "DgElementArray") {
+          str_component_list += " - " + component_name + "\n";
+        }
+      });
+
+  // A list of names was specified
+  if (components_to_monitor.has_value()) {
+    for (const auto& component : *components_to_monitor) {
+      // Do some checks:
+      //  1. Make sure the components requested are viable components. This
+      //     protects against spelling errors.
+      //  2. Currently the only charm Array you can monitor the memory of is
+      //     the DgElementArray so enforce this.
+      if (existing_components.count(component) != 1) {
+        PARSE_ERROR(
+            context,
+            "Cannot monitor memory usage of unknown parallel component '"
+                << component
+                << "'. Please choose from the existing parallel components:\n"
+                << str_component_list);
+      } else if (existing_components.at(component) == "Array" and
+                 component != "DgElementArray") {
+        PARSE_ERROR(
+            context,
+            "Cannot monitor the '"
+                << component
+                << "' parallel component. Currently, the only Array parallel "
+                   "component allowed to be monitored is the "
+                   "DgElementArray.");
+      }
+
+      components_to_monitor_.insert(component);
+    }
+  } else {
+    // 'All' was specified. No need to filter out Array components (except for
+    // DgElementArray) here. That was already done above during the
+    // construction of existing_components
```

Comment:
I don't see any code doing this.  Am I missing something?

Reply:
Oh wait, you're right. I'm mistaken. I'll add in the filter.